roboteam_ai/src/skills/Receive.cpp

```diff
@@ -62,9 +63,37 @@ void Receive::onTerminate(Status s) {
 
 // Pick the closest point to the (predicted) line of the ball for any 'regular' interception
 Vector2 Receive::computeInterceptPoint(const Vector2& startBall, const Vector2& endBall) {
-    return robot->pos.project(startBall, endBall);
-}
+    double margin = 0.3;
+    Vector2 projectPos = Line(startBall,endBall).project(robot->pos);
+    if (canMoveInDefenseArea) {
+        return projectPos;
+    }
+    if (world::field->pointIsInDefenceArea(projectPos, true, margin)||world::field->pointIsInDefenceArea(projectPos, false, margin)) {
+        Polygon dUs(world::field->getDefenseArea(false, margin)), dThem(world::field->getDefenseArea(true, margin));
```

Comment:
`dUs `is a weird name?

Reply:
refactored to defUs

Reply:
dusssss

Reply:
dthemmmm

Reply:
please initialize the two polygons on seperate lines